jaxrs-spec/src/main/asciidoc/chapters/resources/_resource_method.adoc

```diff
@@ -134,34 +134,19 @@ exceptions and process them in the following order:
 
 1.  Instances of `WebApplicationException` and its subclasses MUST be
 mapped to a response as follows. If the `response` property of the
-exception does not contain an entity and an exception mapping provider
-(see <<exceptionmapper>>) is available for
-`WebApplicationException` or the corresponding subclass, an
-implementation MUST use the provider to create a new
-`Response` instance, otherwise the `response` property is used directly.
+exception is null, then an
+implementation MUST use the most appropriate
+exception mapping provider (see <<exceptionmapper>>) to create a new
```

Comment:
A quick question, although it is a bit off-topic. My understanding of the old (and also the new) wording is that implementations MUST NOT invoke exception mappers if a `WebApplicationException` with a non-null `response` property is thrown. Is that correct?

AFAIK Jersey invokes a mapper defined as `ExceptionMapper<Throwable>` for `throw new NotFoundException()` which internally creates a default `response` in the exception constructor.

Reply:
re: "mplementations MUST NOT invoke exception mappers if a WebApplicationException with a non-null response property is thrown"

That's my understanding as well.

re: "Jersey invokes a mapper defined as ExceptionMapper<Throwable>"

I don't think I understand. NotFoundException is a subclass of WebApplicationException, so it should be handled as a WebApplicationException, right?

Reply:
> I don't think I understand. NotFoundException is a subclass of WebApplicationException, so it should be handled as a WebApplicationException, right?

Yes, that's my opinion too. But AFAIK Jersey invokes the ExceptionMapper in this case. However, that's really off-topic and I may even be wrong about this. I'll bring this up to the Jersey list. I was just curious if my understanding of the current wording is correct.